packages/slate-editor/src/extensions/press-contacts/components/PressContactElement.tsx

```diff
@@ -1,32 +1,72 @@
-import type { ContactNode } from '@prezly/slate-types';
-import type { ContactInfo } from '@prezly/slate-types';
+import type { ContactInfo, ContactNode } from '@prezly/slate-types';
+import { ContactLayout } from '@prezly/slate-types';
 import classNames from 'classnames';
-import type { ReactNode } from 'react';
+import { useCallback } from 'react';
 import React from 'react';
+import type { FunctionComponent, ReactNode, SVGProps } from 'react';
 import type { RenderElementProps } from 'slate-react';
+import { useSlateStatic } from 'slate-react';
 
 import { Avatar, EditorBlock } from '#components';
-import { User } from '#icons';
+import { Envelope, Globe, Mobile, Phone, SocialFacebook, SocialTwitter, User } from '#icons';
+
+import { removePressContact, updatePressContact } from '../lib';
 
 import styles from './PressContactElement.module.scss';
+import { PressContactMenu } from './PressContactMenu';
 
 interface Props extends RenderElementProps {
     element: ContactNode;
     renderMenu?: (props: { onClose: () => void }) => ReactNode;
 }
 
 export function PressContactElement({ attributes, children, element, renderMenu }: Props) {
+    const editor = useSlateStatic();
+    const { layout, show_avatar: showAvatar } = element;
+    const isCardLayout = layout === ContactLayout.CARD;
+    const isSignatureLayout = layout === ContactLayout.SIGNATURE;
+
+    const handleToggleAvatar = useCallback(
+        (showAvatar: boolean) => updatePressContact(editor, element, { show_avatar: showAvatar }),
+        [editor, element],
+    );
+    const handleChangeLayout = useCallback(
+        (layout: ContactLayout) => updatePressContact(editor, element, { layout }),
+        [editor, element],
+    );
+    const handleRemove = useCallback(() => removePressContact(editor), [editor, element]);
```

Comment:
Is `element` dependency necessary here?

Reply:
No 🙂 330059c18adbe5564a654ac60a76ddb593412860